app/src/main/java/com/github/featuredetectandroid/utils/KeypointExtractionAlgorithm.kt

```diff
@@ -0,0 +1,31 @@
+package com.github.featuredetectandroid.utils
+
+import android.util.Log
+
+private const val TAG = "KeypointExtractionAlgorithm"
+
+enum class KeypointExtractionAlgorithm {
+    NONE,
+    SIFT,
+    SURF,
+    ORB,
+    SUPERPOINT
+}
+
+fun algorithmsNamesAsList() = enumValues<KeypointExtractionAlgorithm>().map {
+    when (it) {
+        KeypointExtractionAlgorithm.NONE -> "None"
+        KeypointExtractionAlgorithm.SUPERPOINT -> "SuperPoint"
+        else -> it.name
+    }
+}
+
+fun getAlgorithmByName(name: String): KeypointExtractionAlgorithm {
+    var algorithm = KeypointExtractionAlgorithm.NONE
+    try {
+        algorithm = KeypointExtractionAlgorithm.valueOf(name)
+    } catch (illegalAlgorithmName: IllegalArgumentException) {
+        Log.e(TAG, "Illegal algorithm name: $name. $illegalAlgorithmName")
```

Comment:
This `TAG` is too long -- Android Studio gives a error

Reply:
Also, there is syntax for doing what you did here: `Log.e(TAG, "Illegal algorithm name: $name.", illegalAlgorithmName)`

Reply:
And don't give special names for errors in `catch` unless you need to -- let it be just `ex: IllegalArgumentException` or even `e: IllegalArgumentException`

Reply:
I removed this function after discussing the overall vision of the architecture.